ui/src/app/components/sharded-dataflow-migration-details-form/sharded-dataflow-migration-details-form.component.html

```diff
@@ -0,0 +1,205 @@
+                        (change)="onItemChange(option.value, 'source')">
+                        {{option.display}}</mat-radio-button>
+                </mat-radio-group>
+            </div>
+            <br>
+            <div *ngIf="(selectedSourceProfileOption === 'existing')">
+                <mat-form-field appearance="outline">
+                    <mat-label>Select source connection profile</mat-label>
+                    <mat-select class="input-field" formControlName="existingSourceProfile" required="true"
+                        ng-value="selectedProfile">
+                        <mat-option *ngFor="let profile of sourceProfileList" [value]="profile.DisplayName">
+                            {{profile.DisplayName}}
+                        </mat-option>
+                    </mat-select>
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Data Shard Id</mat-label>
+                    <input matInput placeholder="Data ShardId" type="text" formControlName="dataShardId"
+                        required="true" />
+                </mat-form-field>
+            </div>
+            <div *ngIf="(selectedSourceProfileOption === 'new')">
+                <mat-form-field appearance="outline">
+                    <mat-label>Host</mat-label>
+                    <input matInput placeholder="Host" type="text" formControlName="host" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>User</mat-label>
+                    <input matInput placeholder="User" type="text" formControlName="user" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Port</mat-label>
+                    <input matInput placeholder="Port" type="text" formControlName="port" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Password</mat-label>
+                    <input matInput placeholder="Password" type="password" formControlName="password" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Connection profile name</mat-label>
+                    <input matInput placeholder="Connection profile name" type="text" formControlName="newSourceProfile"
+                        required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Data Shard Id</mat-label>
+                    <input matInput placeholder="Data ShardId" type="text" formControlName="dataShardId"
+                        required="true" />
+                </mat-form-field>
+            </div>
+            <div *ngIf="(selectedSourceProfileOption === 'new')">
```

Comment:
Is this check required a second time here ?

Reply:
Not really, both the form and the IP information can be part of the same `div`, I've broken it out into two `divs` since they are doing two different things, for clarity.